standalone/src/test/java/io/pravega/local/PravegaEmulatorResource.java

```diff
@@ -0,0 +1,143 @@
+import org.junit.Assert;
+import org.junit.rules.ExternalResource;
+
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * This class contains the rules to start and stop LocalPravega / standalone.
+ * This resource can be configured for a test using @ClassRule and it will ensure standalone is started once
+ * for all the methods in a test and is shutdown once all the tests methods have completed execution.
+ */
+@Slf4j
+public class PravegaEmulatorResource extends ExternalResource {
+    final boolean restEnabled = false; // disable REST server for the current tests.
+    final boolean authEnabled;
+    final boolean tlsEnabled;
+    final LocalPravegaEmulator pravega;
+
+    public PravegaEmulatorResource(boolean authEnabled, boolean tlsEnabled) {
+        this.authEnabled = authEnabled;
+        this.tlsEnabled = tlsEnabled;
+        LocalPravegaEmulator.LocalPravegaEmulatorBuilder emulatorBuilder = LocalPravegaEmulator.builder()
+                .controllerPort(TestUtils.getAvailableListenPort())
+                .segmentStorePort(TestUtils.getAvailableListenPort())
+                .zkPort(TestUtils.getAvailableListenPort())
+                .restServerPort(TestUtils.getAvailableListenPort())
+                .enableRestServer(restEnabled)
+                .enableAuth(authEnabled)
+                .enableTls(tlsEnabled);
+
+        // Since the server is being built right here, avoiding delegating these conditions to subclasses via factory
+        // methods. This is so that it is easy to see the difference in server configs all in one place. This is also
+        // unlike the ClientConfig preparation which is being delegated to factory methods to make their preparation
+        // explicit in the respective test classes.
+
+        if (authEnabled) {
+            emulatorBuilder.passwdFile(SecurityConfigDefaults.AUTH_HANDLER_INPUT_PATH)
+                    .userName(SecurityConfigDefaults.AUTH_ADMIN_USERNAME)
+                    .passwd(SecurityConfigDefaults.AUTH_ADMIN_PASSWORD);
+        }
+        if (tlsEnabled) {
+            emulatorBuilder.certFile(SecurityConfigDefaults.TLS_SERVER_CERT_PATH)
+                    .keyFile(SecurityConfigDefaults.TLS_SERVER_PRIVATE_KEY_PATH)
+                    .jksKeyFile(SecurityConfigDefaults.TLS_SERVER_KEYSTORE_PATH)
+                    .jksTrustFile(SecurityConfigDefaults.TLS_CLIENT_TRUSTSTORE_PATH)
+                    .keyPasswordFile(SecurityConfigDefaults.TLS_PASSWORD_PATH);
+        }
+
+        pravega = emulatorBuilder.build();
+    }
+
+    protected void before() throws Exception {
+        pravega.start();
+    }
+
+    @SneakyThrows
+    protected void after() {
+        pravega.close();
+    }
+
+    public void testWriteAndReadAnEventWithValidClientConfig(String scope, String streamName, String message, ClientConfig clientConfig) throws Exception {
```

Comment:
It seems like this class could be used by more than one test. Putting this here undermines that.

Reply:
Fixed it.